uni/lib/view/course_unit_info/widgets/course_unit_files_row.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:provider/provider.dart';
+import 'package:uni/controller/local_storage/file_offline_storage.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni/model/entities/course_units/course_unit_file.dart';
+import 'package:uni/model/providers/startup/session_provider.dart';
+import 'package:uni/view/common_widgets/toast_message.dart';
+
+class CourseUnitFilesRow extends StatefulWidget {
+  const CourseUnitFilesRow(this.file, {super.key});
+
+  final CourseUnitFile file;
+
+  @override
+  State<StatefulWidget> createState() {
+    return CourseUnitFilesRowState();
+  }
+}
+
+class CourseUnitFilesRowState extends State<CourseUnitFilesRow>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 1500),
+      vsync: this,
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.only(bottom: 20),
+      child: Row(
+        children: [
+          const SizedBox(width: 8),
+          const Icon(Icons.picture_as_pdf),
+          const SizedBox(width: 1),
+          Expanded(
+            child: GestureDetector(
+              onTap: () {
+                _controller
+                  ..reset()
+                  ..repeat(reverse: true);
+                openFile(context, widget.file);
+              },
+              child: Container(
+                padding: const EdgeInsets.only(left: 10),
+                child: AnimatedBuilder(
+                  animation: _controller,
+                  builder: (context, child) {
+                    return Opacity(
+                      opacity: 1 - 0.5 * sin(_controller.value * pi),
+                      child: Text(
+                        widget.file.name
+                            .substring(0, widget.file.name.indexOf('_')),
+                        overflow: TextOverflow.ellipsis,
+                        style: Theme.of(context).textTheme.bodyLarge,
+                      ),
+                    );
+                  },
+                ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Future<void> openFile(BuildContext context, CourseUnitFile unitFile) async {
+    final session = context.read<SessionProvider>().state!;
+
+    try {
+      final result = await loadFileFromStorageOrRetrieveNew(
+        unitFile.name,
+        unitFile.url,
+        session,
+        headers: {'pct_id': unitFile.fileCode},
+      );
+      await OpenFile.open(result!.path);
+    } catch (e) {
+      if (context.mounted) {
+        await ToastMessage.error(
+          context,
+          S.of(context).load_error,
+        );
+      }
+    }
+    _controller.reset();
+  }
+}
```

Comment:
@DGoiana update: I think it was because I don't have any compatible installed. You should handle the open result:
```
      final openResult = await OpenFile.open(file!.path);
      
      switch (openResult.type) {
        case ResultType.done:
          // TODO: Handle this case.
        case ResultType.fileNotFound:
          // TODO: Handle this case.
        case ResultType.noAppToOpen:
          // TODO: Handle this case.
        case ResultType.permissionDenied:
          // TODO: Handle this case.
        case ResultType.error:
          // TODO: Handle this case.
      }
```
Make sure to encapsulate this behaviour, like `url_toast`